types/decimal.go

```diff
@@ -366,7 +368,8 @@ func (d Dec) ApproxRoot(root uint64) (guess Dec, err error) {
 	rootInt := NewIntFromUint64(root)
 	guess, delta := OneDec(), OneDec()
 
-	for delta.Abs().GT(SmallestDec()) {
+	iter := 0
+	for delta.Abs().GT(SmallestDec()) && iter < 100 {
```

Comment:
```suggestion
	for iter := 0; delta.Abs().GT(SmallestDec()) && iter < 100; iter++ {
```

Reply:
and maybe `100` could be stored in a private const

Reply:
...like `maxApproxRootIterations`?

Reply:
Ok will do that

Reply:
It's funny that's what I did at first but changed it because I thought it was extra 😆